sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -285,6 +290,9 @@ private[sql] case class PreWriteCheck(conf: SQLConf, catalog: SessionCatalog)
 
   def apply(plan: LogicalPlan): Unit = {
     plan.foreach {
+      case CreateTable(tableDesc, _, Some(_))
+          if tableDesc.provider.get == "hive" && sparkConf.get(CATALOG_IMPLEMENTATION) != "hive" =>
```

Comment:
Yeah, you are right. We did it for a few cases, but we are not able to handle all the cases. See the following code: https://github.com/apache/spark/blob/5959df217df53196607b7fa744cdc2b36311360e/sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala#L1017-L1030

We can extend it to support more cases, like another PR: https://github.com/apache/spark/pull/13907. However, this sounds like we are unable to support all the cases when Hive is not enabled. 

Regarding the issue you mentioned above, (reading the schema from the existing Hive table and then append to another table), there are two cases:

- When Hive support is enabled, another fix of the PR https://github.com/apache/spark/pull/14410 is trying to resolve it. Will submit it soon. 

- When Hive support is not enabled, reading Hive metastore is impossible. Thus, should we still issue an exception? 

Sorry, this also looks very confusing to me. Hopefully, my description is a little bit clear.